crates/network-primitives/src/traits.rs

```diff
@@ -46,6 +48,32 @@ pub trait TransactionResponse {
     fn input(&self) -> &Bytes;
 }
 
+/// Header JSON-RPC response.
+pub trait HeaderResponse {
+    /// Base fee per unit of gas (If EIP-1559 is supported)
+    fn base_fee_per_gas(&self) -> Option<u128>;
+
+    /// Blob fee for the next block (if EIP-4844 is supported)
+    fn next_block_blob_fee(&self) -> Option<u128>;
```

Comment:
this is a good start, we can add more functions based on the spec:

https://ethereum.github.io/execution-apis/api-documentation/